nav2_simple_commander/nav2_simple_commander/demo_recoveries.py

```diff
@@ -0,0 +1,104 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from geometry_msgs.msg import PoseStamped
+from nav2_simple_commander.robot_navigator import BasicNavigator, NavigationResult
+import rclpy
+from rclpy.duration import Duration
+
+
+"""
+Basic stock inspection demo. In this demonstration, the expectation
+is that there are cameras or RFID sensors mounted on the robots
+collecting information about stock quantity and location.
+"""
+
+
+def main():
+    rclpy.init()
+
+    navigator = BasicNavigator()
+
+    # Set our demo's initial pose
+    initial_pose = PoseStamped()
+    initial_pose.header.frame_id = 'map'
+    initial_pose.header.stamp = navigator.get_clock().now().to_msg()
+    initial_pose.pose.position.x = 3.45
+    initial_pose.pose.position.y = 2.15
+    initial_pose.pose.orientation.z = 1.0
+    initial_pose.pose.orientation.w = 0.0
+    navigator.setInitialPose(initial_pose)
+
+    # Wait for navigation to fully activate
+    navigator.waitUntilNav2Active()
+
+    goal_pose = PoseStamped()
+    goal_pose.header.frame_id = 'map'
+    goal_pose.header.stamp = navigator.get_clock().now().to_msg()
+    goal_pose.pose.position.x = 6.13
+    goal_pose.pose.position.y = 1.90
+    goal_pose.pose.orientation.w = 1.0
+
+    navigator.goToPose(goal_pose)
+
+    i = 0
+    while not navigator.isNavComplete():
+        i += 1
+        feedback = navigator.getFeedback()
+        if feedback and i % 5 == 0:
+            print(
+                f'Estimated time of arrival to destination is: \
+                {Duration.from_msg(feedback.estimated_time_remaining).nanoseconds / 1e9}'
+            )
+
+    # Robot hit a dead end, back it up
+    navigator.backup(backup_dist=0.5, backup_speed=0.1)
+
+    i = 0
+    while not navigator.isNavComplete():
+        i += 1
+        feedback = navigator.getFeedback()
+        if feedback and i % 5 == 0:
+            print(f'Distance traveled: {feedback.distance_traveled}')
+
+    # Turn it around
+    navigator.spin(spin_dist=3.14)
+
+    i = 0
+    while not navigator.isNavComplete():
+        i += 1
+        feedback = navigator.getFeedback()
+        if feedback and i % 5 == 0:
+            print(f'Spin angle traveled: {feedback.angular_distance_traveled}')
+
+    result = navigator.getResult()
+    if result == NavigationResult.SUCCEEDED:
+        print('Dead end confirmed! Returning to start...')
+    elif result == NavigationResult.CANCELED:
+        print('Recovery was canceled. Returning to start...')
+        exit(1)
```

Comment:
Exit(1) doesn't look like returning to start :wink:

Reply:
Looks like you found another bug 
https://github.com/ros-planning/navigation2/blob/d83f9d995326f3aa1fb76c7cbe879ab783d95955/nav2_simple_commander/nav2_simple_commander/demo_inspection.py#L87

Will fix it.